sml/src/sml_transport.c

```diff
@@ -34,17 +35,21 @@ unsigned char end_seq[] = {0x1b, 0x1b, 0x1b, 0x1b, 0x1a};
 
 
 size_t sml_read(int fd, fd_set *set, unsigned char *buffer, size_t len) {
-	
+
 	ssize_t r = 0;
 	size_t tr = 0;
 
 	while (tr < len) {
 		select(fd + 1, set, 0, 0, 0);
 		if (FD_ISSET(fd, set)) {
-			
-			r = read(fd, &(buffer[tr]), len - tr);
-			if (r < 0) continue;
 
+			r = read(fd, &(buffer[tr]), len - tr);
+			if (r == 0) return 0; // EOF
+			if (r == EINTR || r == EAGAIN) continue; // should be ignored
+			if (r < 0) {
+				fprintf(stderr, "libsml: sml_read(): read error\n");
```

Comment:
would be useful to print errno as well.

Reply:
see https://github.com/volkszaehler/libsml/pull/51#issuecomment-436005161